public/src/app.js

```diff
@@ -106,6 +104,31 @@ app.cacheBuster = null;
 		});
 	};
 
+	app.updateHeader = function (data, callback) {
+		/**
+		 * data:
+		 *   header (obj)
+		 *   config (obj)
+		 *   next (string)
+		 */
+		require(['benchpress'], function (Benchpress) {
+			app.user = data.header.user;
+			data.header.config = data.config;
+			config = data.config;
+			Benchpress.setGlobal('config', config);
+
+			// Manually reconnect socket.io
+			socket.close();
+			socket.open();
+
+			// Re-render top bar menu
+			app.parseAndTranslate('partials/menu', data.header, function (html) {
+				$('#header-menu .container').html(html);
```

Comment:
AFAIK persona has its own mobile menu that is cloned from the actual menu, does it get updated when someone logs in?